common/src/responder_id.rs

```diff
@@ -24,28 +25,80 @@ pub enum ResponderIdParseError {
 }
 
 /// Node unique identifier.
-#[derive(
-    Clone, Default, Debug, Eq, Serialize, Deserialize, PartialEq, PartialOrd, Ord, Hash, Digestible,
-)]
-pub struct ResponderId(pub String);
+#[derive(Clone, Default, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, Digestible)]
+pub struct ResponderId {
+    /// The node's host name. See https://tools.ietf.org/html/rfc1123
+    pub host: String,
+    /// The node's port.
+    pub port: u16,
+}
+
+impl ResponderId {
+    /// Create a new ResponderId.
+    ///
+    /// # Arguments
+    /// * `host` - Node's host name
+    /// * `port` - Node's port number
+    pub fn new(host: &str, port: u16) -> Result<Self, ResponderIdParseError> {
+        let host_port = alloc::format!("{}:{}", host, port);
+        ResponderId::from_str(&host_port)
+    }
+}
+
+impl Serialize for ResponderId {
+    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&alloc::format!("{}:{}", &self.host, &self.port))
+    }
+}
+
+impl<'de> Deserialize<'de> for ResponderId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        ResponderId::from_str(&s).map_err(|_e| D::Error::custom("Invalid ResponderId"))
+    }
+}
 
 impl Display for ResponderId {
     fn fmt(&self, f: &mut Formatter) -> FmtResult {
-        write!(f, "{}", self.0)
+        write!(f, "{}:{}", self.host, self.port)
     }
 }
 
 impl FromStr for ResponderId {
     type Err = ResponderIdParseError;
 
     fn from_str(src: &str) -> Result<ResponderId, Self::Err> {
-        // ResponderId is expected to be host:port, so at least ensure we have a single colon as a
-        // small sanity test.
-        if !src.contains(':') {
-            return Err(ResponderIdParseError::InvalidFormat(src.to_string()));
-        }
+        // host:port or host:port/, with named capture groups.
+        let re = Regex::new(r#"(?P<host>^[^:]+):(?P<port>[0-9]{1,5})/?"#).unwrap();
```

Comment:
Feedback on this regex is welcome

Reply:
I think a trailing slash should never find its way to the ResponderId so we shouldn't allow that.